packages/yourloops/test/unit/components/patient/patient-row.test.tsx

```diff
@@ -32,69 +32,75 @@ import * as authHookMock from '../../../../lib/auth'
 import { User } from '../../../../lib/auth'
 import { createPatient, createPatientTeam } from '../../common/utils'
 import PatientRow from '../../../../components/patient/patient-row'
-import { PatientElementProps } from '../../../../components/patient/models'
+import { PatientRowProps } from '../../../../components/patient/models'
 import { fireEvent, render, screen, within } from '@testing-library/react'
 import Table from '@material-ui/core/Table'
 import TableBody from '@material-ui/core/TableBody'
 import userEvent from '@testing-library/user-event'
 import PatientUtils from '../../../../lib/patient/utils'
 
+const removePatientDialogMockId = 'remove-patient-dialog-id'
+// eslint-disable-next-line react/display-name
+jest.mock('../../../../components/patient/remove-patient-dialog', () => () => {
+  return <div data-testid={removePatientDialogMockId} />
+})
 jest.mock('../../../../lib/auth')
 describe('Patient row', () => {
-  const onClickPatient = jest.fn()
-  const onFlagPatient = jest.fn()
-
+  const flagPatientMock = jest.fn()
   const teamId = 'teamId'
   const teams = [createPatientTeam(teamId, UserInvitationStatus.accepted)]
   const patient = createPatient('id1', teams)
-  const props: PatientElementProps = {
+  const props: PatientRowProps = {
     patient,
-    flagged: [],
-    filter: undefined,
-    onClickPatient,
-    onFlagPatient
+    filter: undefined
   }
 
   beforeEach(() => {
     (authHookMock.useAuth as jest.Mock) = jest.fn().mockImplementation(() => {
-      return { user: { isUserHcp: () => true } as User }
+      return {
+        flagPatient: flagPatientMock,
+        getFlagPatients: jest.fn().mockReturnValue([]),
+        user: { isUserHcp: () => true } as User
+      }
     })
   })
 
-  const getPatientRowJSX = (patientElementProps: PatientElementProps = props): JSX.Element => {
+  const getPatientRowJSX = (patientElementProps: PatientRowProps = props): JSX.Element => {
     return (
       <Table>
         <TableBody>
           <PatientRow
             patient={patientElementProps.patient}
-            flagged={patientElementProps.flagged}
             filter={patientElementProps.filter}
-            onClickPatient={patientElementProps.onClickPatient}
-            onFlagPatient={patientElementProps.onFlagPatient}
           />
         </TableBody>
       </Table>
     )
   }
 
-  it('clicking on flag icon should flag patient', () => {
+  it('should open modal when clicking on remove patient icon', () => {
```

Comment:
From what came out of the react training, this should be done in an IT. I can help you with it.